pkg/probe/option.go

```diff
@@ -27,6 +30,15 @@ type Options struct {
 }
 
 // IsValid returns true if some values are filled into the options.
-func (o *Options) IsValid() bool {
-	return o != nil && o.Path != "" && o.UpdateInterval > 0*time.Second
+func (o *Options) Validate() error {
+	if o == nil {
+		return fmt.Errorf("option is nil")
```

Comment:
`fmt.Errorf` isn't needed here, as you aren't doing any formatting. `errors.New` would suffice.

Reply:
@douglas-reid fixed.